popup_survey.module

```diff
@@ -174,12 +176,15 @@ function popup_survey_bean_options() {
     ->entityCondition('entity_type', 'bean', '=')
     ->propertyCondition('type', 'survey_popup_message', '=');
   $result = $query->execute();
-  
+
   // From the query result pull out each of the bean ids.
   $ids = array();
-  foreach($result['bean'] as $record) {
-    $ids[] = $record->bid;
+  if( count($result) > 0 ) {
```

Comment:
I tried isset first and then is_array but the query result is always an array, just an empty one so they  always returns true.  This is a check for the existence of data, not a failed query.